src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -1283,14 +1283,9 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]) -> None:
         self.mgr.trigger_connect_dashboard_rgw()
 
     def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[str, Any], List[str]]:
-        config, deps = super().generate_config(daemon_spec)
-        rgw_spec = cast(RGWSpec, self.mgr.spec_store[daemon_spec.service_name].spec)
-        ssl_cert = getattr(rgw_spec, 'rgw_frontend_ssl_certificate', None)
-        if isinstance(ssl_cert, list):
-            ssl_cert = '\n'.join(ssl_cert)
-        if ssl_cert:
-            deps.append(str(utils.md5_hash(ssl_cert)))
-        return config, deps
+        svc_spec = cast(RGWSpec, self.mgr.spec_store[daemon_spec.service_name].spec)
+        config, _ = super().generate_config(daemon_spec)
+        return config, self.get_dependencies(self.mgr, svc_spec)
```

Comment:
I don't think it has an effect right now, but `generally I'd prefer if we take the deps from the parent class `generate_config` and then append the deps from our specific class

Reply:
that's a valid point, I fixed the code.